app/src/test/java/org/simple/clinic/medicalhistory/newentry/NewMedicalHistoryScreenLogicTest.kt

```diff
@@ -84,10 +84,23 @@ class NewMedicalHistoryScreenLogicTest {
   fun `when save is clicked with selected answers then patient with the answers should be saved and summary screen should be opened`() {
     // given
     val savedPatient = TestData.patient(uuid = patientUuid)
-    whenever(patientRepository.saveOngoingEntryAsPatient(eq(user), eq(facility), any(), any(), any())).thenReturn(Single.just(savedPatient))
+    val addressUuid = UUID.fromString("79f6d1bc-0e10-480b-9cce-43c628d827b7")
+
+    val uuidGenerator = mock<UuidGenerator>()
+    whenever(uuidGenerator.v4()).thenReturn(patientUuid, addressUuid, medicalHistoryUuid)
+
+    whenever(patientRepository.saveOngoingEntryAsPatient(
+        loggedInUser = eq(user),
+        facility = eq(facility),
+        patientUuid = eq(patientUuid),
+        addressUuid = eq(addressUuid),
+        supplyUuidForBpPassport = any(),
```

Comment:
We cannot do this here since the `supplyUuid` parameters are functions and not values. This will be fixed when we refactor the patient entry flow (See other comment for a link to the ticket).